pkg/probe/http/http_test.go

```diff
@@ -87,13 +90,156 @@ func TestHTTPProbeProxy(t *testing.T) {
+			w.WriteHeader(s)
+			w.Write([]byte(body))
+		}
+	}
+	followNonLocalRedirects := true
+	prober := New(followNonLocalRedirects)
+
+	testCases := []struct {
+		handler           func(w http.ResponseWriter, r *http.Request)
+		reqHeaders        http.Header
+		health            probe.Result
+		expectHTTPCodes   []int
+		expectHTTPContent string
+		accBody           string
+		notBody           string
+	}{
+		// The probe will be filled in below.  This is primarily testing that an HTTP GET happens.
+		{
+			handler:           handleReq(http.StatusOK, "helloworld"),
+			expectHTTPCodes:   []int{},
+			expectHTTPContent: "helloworld",
+			health:            probe.Success,
+		},
+		{
+			handler:           handleReq(http.StatusOK, "ok"),
+			expectHTTPCodes:   []int{},
+			expectHTTPContent: "",
+			health:            probe.Success,
+		},
+		{
+			handler:           handleReq(http.StatusNotFound, "not found"),
+			expectHTTPCodes:   []int{},
+			expectHTTPContent: "",
+			health:            probe.Failure,
+		},
+		{
+			handler:           handleReq(http.StatusNotFound, "not found"),
+			expectHTTPCodes:   []int{404},
+			expectHTTPContent: "",
+			health:            probe.Success,
+		},
+		{
+			handler:           handleReq(http.StatusNotFound, "not found"),
+			expectHTTPCodes:   []int{500, 404},
+			expectHTTPContent: "",
+			health:            probe.Success,
+		},
+		{
+			handler:           handleReq(http.StatusNotFound, "not found"),
+			expectHTTPCodes:   []int{200},
+			expectHTTPContent: "",
+			health:            probe.Failure,
+		},
+		{
+			handler:           handleReq(http.StatusNotFound, "not found"),
+			expectHTTPCodes:   []int{400, 200},
+			expectHTTPContent: "",
+			health:            probe.Failure,
+		},
+		{
+			handler:           handleReq(http.StatusNotFound, "not found"),
+			expectHTTPCodes:   []int{404, 500},
+			expectHTTPContent: "not*",
+			health:            probe.Success,
+		},
+		{
+			handler:           handleReq(http.StatusNotFound, "not found"),
+			expectHTTPCodes:   []int{404, 500},
+			expectHTTPContent: "found*",
+			health:            probe.Failure,
+		},
+		{
+			handler:           handleReq(http.StatusNotFound, "not found"),
+			expectHTTPCodes:   []int{200},
+			expectHTTPContent: "not *",
+			health:            probe.Failure,
+		},
+		{
+			handler:           handleReq(http.StatusNotFound, "not found"),
+			expectHTTPCodes:   []int{404, 500},
+			expectHTTPContent: "not *",
+			health:            probe.Success,
+		},
+		{
+			handler:           handleReq(http.StatusNotFound, "not found"),
+			expectHTTPCodes:   []int{},
+			expectHTTPContent: "not *",
+			health:            probe.Success,
+		},
+		{
+			handler:           handleReq(http.StatusNotFound, "not found"),
+			expectHTTPCodes:   []int{},
+			expectHTTPContent: "found *",
+			health:            probe.Failure,
+		},
+	}
+
+	for i, test := range testCases {
+		t.Run(fmt.Sprintf("case-%2d", i), func(t *testing.T) {
```

Comment:
Would it be more readable if we introduce a struct with test cases will have a member with a human-readable name that we could use here (instead of `case-N`)?

Reply:
I add the name for the each test like this: 
`t.Run(fmt.Sprintf("case-%2d: assert expectHTTPCodes='%v', expectHTTPContent='%s' for response body='%s', statusCode='%d' is %s", i, test.expectHTTPCodes, test.expectHTTPContent, test.body, test.statusCode, test.health)`

how about this?